core/src/test/java/org/opensearch/sql/expression/datetime/TimeStampAddTest.java

```diff
@@ -0,0 +1,197 @@
+import org.opensearch.sql.data.model.ExprStringValue;
+import org.opensearch.sql.data.model.ExprTimeValue;
+import org.opensearch.sql.data.model.ExprTimestampValue;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.exception.SemanticCheckException;
+import org.opensearch.sql.expression.DSL;
+import org.opensearch.sql.expression.Expression;
+import org.opensearch.sql.expression.ExpressionTestBase;
+import org.opensearch.sql.expression.FunctionExpression;
+
+class TimeStampAddTest extends ExpressionTestBase {
+
+  private static Stream<Arguments> getTestDataForTimestampAdd() {
+    return Stream.of(
+        Arguments.of("MINUTE", 1, new ExprStringValue("2003-01-02 00:00:00"),
+            "2003-01-02 00:01:00"),
+        Arguments.of("WEEK", 1, new ExprStringValue("2003-01-02 00:00:00"),
+            "2003-01-09 00:00:00"),
+        //Date
+        Arguments.of("MINUTE", 1, new ExprDateValue("2003-01-02"),
+            "2003-01-02 00:01:00"),
+        Arguments.of("WEEK", 1, new ExprDateValue("2003-01-02"),
+            "2003-01-09 00:00:00"),
+        //Datetime
+        Arguments.of("MINUTE", 1, new ExprDatetimeValue("2003-01-02 00:00:00"),
+            "2003-01-02 00:01:00"),
+        Arguments.of("WEEK", 1, new ExprDatetimeValue("2003-01-02 00:00:00"),
+            "2003-01-09 00:00:00"),
+        //timstamp
+        Arguments.of("MINUTE", 1, new ExprTimestampValue("2003-01-02 00:00:00"),
+            "2003-01-02 00:01:00"),
+        Arguments.of("WEEK", 1, new ExprTimestampValue("2003-01-02 00:00:00"),
+            "2003-01-09 00:00:00"),
+        //Cases surrounding leap year
+        Arguments.of("SECOND", 1, new ExprTimestampValue("2020-02-28 23:59:59"),
+            "2020-02-29 00:00:00"),
+        Arguments.of("MINUTE", 1, new ExprTimestampValue("2020-02-28 23:59:59"),
+            "2020-02-29 00:00:59"),
+        Arguments.of("HOUR", 1, new ExprTimestampValue("2020-02-28 23:59:59"),
+            "2020-02-29 00:59:59"),
+        Arguments.of("DAY", 1, new ExprTimestampValue("2020-02-28 23:59:59"),
+            "2020-02-29 23:59:59"),
+        Arguments.of("WEEK", 1, new ExprTimestampValue("2020-02-28 23:59:59"),
+            "2020-03-06 23:59:59"),
+        //Cases surrounding end-of-year
+        Arguments.of("SECOND", 1, new ExprTimestampValue("2020-12-31 23:59:59"),
+            "2021-01-01 00:00:00"),
+        Arguments.of("MINUTE", 1, new ExprTimestampValue("2020-12-31 23:59:59"),
+            "2021-01-01 00:00:59"),
+        Arguments.of("HOUR", 1, new ExprTimestampValue("2020-12-31 23:59:59"),
+            "2021-01-01 00:59:59"),
+        Arguments.of("DAY", 1, new ExprTimestampValue("2020-12-31 23:59:59"),
+            "2021-01-01 23:59:59"),
+        Arguments.of("WEEK", 1, new ExprTimestampValue("2020-12-31 23:59:59"),
+            "2021-01-07 23:59:59"),
+
+        //Test remaining interval types
+        Arguments.of("MICROSECOND", 1, new ExprStringValue("2003-01-02 00:00:00"),
```

Comment:
Need to test adding a month (see special cases given in example on [MySQL doc](https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_timestampadd))
Test negative values and values > 1[00] too

Reply:
Added in https://github.com/Bit-Quill/opensearch-project-sql/pull/246/commits/77226baed22abe00079b735df28ff26f2cc7547e